src/exchange.rs

```diff
@@ -22,6 +23,20 @@ impl<'a, T: FormattableCurrency> Exchange<'a, T> {
         self.map.insert(key, *rate);
     }
 
+    pub fn set_rate_and_inverse(&mut self, rate: &'a ExchangeRate<T>) -> Result<ExchangeRate<'a, T>, MoneyError> {
+        if rate.rate == dec!(0) {
+            return Err(MoneyError::DivideByZero)
```

Comment:
How would you like to see divide by zero handled? I felt like returning a `MoneyError` made the most sense but I'm curious to hear your thoughts.